src/logging.cc

```diff
@@ -37,20 +37,40 @@
 #include <unistd.h>
 #endif
 #include <algorithm>
+#include <fstream>
 #include <iomanip>
 #include <iostream>
 
 namespace triton { namespace common {
 
 Logger gLogger_;
 
-Logger::Logger() : enables_{true, true, true}, vlevel_(0), format_(Format::kDEFAULT) {}
+Logger::Logger()
+    : enables_{true, true, true}, vlevel_(0), format_(Format::kDEFAULT),
+      filename_("")
+{
+}
 
 void
 Logger::Log(const std::string& msg)
 {
   const std::lock_guard<std::mutex> lock(mutex_);
-  std::cerr << msg << std::endl;
+  if (Logger::GetLogOutFile() != "") {
+    try {
+      std::ofstream file_stream;
+      file_stream.open(filename_, std::ios::app);
+      file_stream << msg << std::endl;
+      file_stream.close();
```

Comment:
Any reason not to keep the file stream open until it is asked to log to different location?

Reply:
>suggesting a new member variable that holds the stream open once invoked until the out file has been changed

Yes.